features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Operations/Operations.jsx

```diff
@@ -0,0 +1,444 @@
+        display: 'flex',
+        flexWrap: 'wrap',
+    },
+    textField: {
+        marginLeft: theme.spacing.unit,
+        marginRight: theme.spacing.unit,
+        width: 400,
+    },
+    mainTitle: {
+        paddingLeft: 0,
+    },
+    scopes: {
+        width: 400,
+    },
+    titleWrapper: {
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+    },
+    button: {
+        marginLeft: theme.spacing.unit * 2,
+        textTransform: theme.custom.leftMenuTextStyle,
+        color: theme.palette.getContrastText(theme.palette.primary.main),
+    },
+    buttonMain: {
+        textTransform: theme.custom.leftMenuTextStyle,
+        color: theme.palette.getContrastText(theme.palette.primary.main),
+    },
+    addNewWrapper: {
+        backgroundColor: theme.palette.background.paper,
+        color: theme.palette.getContrastText(theme.palette.background.paper),
+        border: 'solid 1px ' + theme.palette.grey['300'],
+        borderRadius: theme.shape.borderRadius,
+        marginTop: theme.spacing.unit * 2,
+    },
+    contentWrapper: {
+        maxWidth: theme.custom.contentAreaWidth,
+    },
+    addNewHeader: {
+        padding: theme.spacing.unit * 2,
+        backgroundColor: theme.palette.grey['300'],
+        fontSize: theme.typography.h6.fontSize,
+        color: theme.typography.h6.color,
+        fontWeight: theme.typography.h6.fontWeight,
+    },
+    addNewOther: {
+        padding: theme.spacing.unit * 2,
+    },
+    radioGroup: {
+        display: 'flex',
+        flexDirection: 'row',
+        width: 300,
+    },
+    addResource: {
+        width: 600,
+        marginTop: 0,
+    },
+    buttonIcon: {
+        marginRight: 10,
+    },
+    expansionPanel: {
+        marginBottom: theme.spacing.unit,
+    },
+    expansionPanelDetails: {
+        flexDirection: 'column',
+    },
+});
+
+/**
+ * This class defined for operation List
+ */
+class Operations extends React.Component {
+    /**
+     *
+     * @param {*} props the props parameters
+     */
+    constructor(props) {
+        super(props);
+        this.state = {
+            swagger: {},
+            scopes: [],
+            notFound: false,
+            showScopes: false,
+            apiPolicies: [],
+            operationList: null,
+        };
+
+        this.newApi = new Api();
+        this.handleUpdateList = this.handleUpdateList.bind(this);
+        this.toggleAssignScopes = this.toggleAssignScopes.bind(this);
+        this.handleScopeChange = this.handleScopeChange.bind(this);
+        this.updateOperations = this.updateOperations.bind(this);
+        this.handleScopeChangeInSwaggerRoot = this.handleScopeChangeInSwaggerRoot.bind(this);
+    }
+
+    /**
+     *
+     */
+    componentDidMount() {
+        this.setState({
+            operationList: this.props.api.operations,
+            scopes: this.props.api.scopes,
+        });
+        const promisedApi = this.newApi.getSwagger(this.props.api.id);
```

Comment:
why are we setting the state at the start of the didMount ? we can use de structuring here too